boot.py

```diff
@@ -190,27 +195,37 @@ def on_post_window_command(self, window: sublime.Window, command_name: str, args
                 sublime.set_timeout(panel_manager.update_log_panel)
 
 
-class LspOpenLocationCommand(sublime_plugin.TextCommand):
+class LspOpenLocationCommand(LspWindowCommand):
     """
     A command to be used by third-party ST packages that need to open an URI with some abstract scheme.
     """
 
     def run(
         self,
-        _: sublime.Edit,
         location: Union[Location, LocationLink],
         session_name: Optional[str] = None,
         flags: int = 0,
-        group: int = -1
+        group: int = -1,
+        event: Optional[dict] = None
     ) -> None:
+        if event:
+            modifier_keys = event.get('modifier_keys')
+            if modifier_keys:
+                if 'primary' in modifier_keys:
+                    flags |= sublime.ADD_TO_SELECTION | sublime.SEMI_TRANSIENT | sublime.CLEAR_TO_RIGHT
+                elif 'shift' in modifier_keys:
+                    flags |= sublime.ADD_TO_SELECTION | sublime.SEMI_TRANSIENT
         sublime.set_timeout_async(lambda: self._run_async(location, session_name, flags, group))
 
+    def want_event(self) -> bool:
+        return True
+
     def _run_async(
-        self, location: Union[Location, LocationLink], session_name: Optional[str], flags: int = 0, group: int = -1
+        self, location: Union[Location, LocationLink], session_name: Optional[str], flags: int, group: int
     ) -> None:
-        manager = windows.lookup(self.view.window())
-        if manager:
-            manager.open_location_async(location, session_name, self.view, flags, group) \
```

Comment:
I would now remove the `manager.open_location_async` implementation since it doesn't appear to be used anymore (even within third-party packages).